### DISCUSSION TITLE
Dependency Injection shortcut with BaseSettings raises fastapi.exceptions.FastAPIError: Invalid args for response field!

### DISCUSSION THREAD

**Discussion by followben** [fastapi/fastapi#9401] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, Depends
from pydantic import BaseSettings, BaseModel

app = FastAPI()

class MyPydanticSettings(BaseSettings):
    foo: str = "foo"

class MyPydanticModel(BaseModel):
    bar: str = "bar"

@app.get("/")
def foobar(
    # Either of these dependencies work fine
    # model: Annotated[MyPydanticModel, Depends()],
    model: MyPydanticModel = Depends(),

    # And either of these will work ok too
    # settings: MyPydanticSettings = Depends(lambda: MyPydanticSettings()),
    # settings: Annotated[MyPydanticSettings, Depends(lambda: MyPydanticSettings())],

    # But neither of these do - both raise a FastAPIError
    settings: MyPydanticSettings = Depends(),
    # settings: Annotated[MyPydanticSettings, Depends()],
):
    return {settings.foo: model.bar}
```

### Description

On starting the application (e.g. `uvicorn main:app --reload`), the following error is raised:

```
fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that typing.Union[str, os.PathLike, typing.List[typing.Union[str, os.PathLike]], typing.Tuple[typing.Union[str, os.PathLike], ...], NoneType] is a valid pydantic field type
```

### Operating System

Other

### Operating System Details

Tested on macOS Ventura and Ubuntu 22.04.2 LTS (Jammy Jellyfish)

### FastAPI Version

0.95.1

### Python Version

3.10.9

### Additional Context

I'm following the [shortcut example from the documentation](https://fastapi.tiangolo.com/tutorial/dependencies/classes-as-dependencies/#shortcut). I've tested as far back as FastAPI 0.75.0 thinking this might have been a regression, but seems injecting Pydantic Settings in this manner has always failed. I guess in the past I've [always used a cached function](https://fastapi.tiangolo.com/advanced/settings/?h=lru_#creating-the-settings-only-once-with-lru_cache) or a module member.

If this is expected behaviour (perhaps due to way BaseSettings get initialised), would it be worth mentioning in the shortcut docs and/ or catching and raising with a more helpful error?

**Comment by mobraine** [comment]
Try to override the __init__ function inside your class, this worked for me.

```
class MyPydanticSettings(BaseSettings):
    foo: str = "foo"

    def __init__(self):
        """Return a new config object instance."""
        super().__init__()

**Comment by alv2017** [comment, marked as answer]
@followben
I think, that you need either instantiate your models, 
or use the approach proposed in [Classes as Dependencies](https://fastapi.tiangolo.com/tutorial/dependencies/classes-as-dependencies/) as @mobraine says

this is for pydantic-settings: https://fastapi.tiangolo.com/advanced/settings/#the-main-app-file
